boards: xtensa: add Olimex ESP32-EVB board definition

henrikbrixandersen opened the pull request:
- Add a GPIO pass-thru map for accessing the full range (0 to 39) of ESP32 GPIO pins by their datasheet number. GPIOs 0 to 31 are mapped to gpio0 while GPIOs 32 to 39 are mapped to gpio1.
- Add board definition for the Olimex ESP32-EVB development board.

henrikbrixandersen committed:
dts: xtensa: espressif: esp32: add GPIO map for accessing full GPIO range

Add a GPIO pass-thru map for accessing the full range (0 to 39) of ESP32
GPIO pins by their datasheet number.

GPIOs 0 to 31 are mapped to gpio0 while GPIOs 32 to 39 are mapped to
gpio1.

Signed-off-by: Henrik Brix Andersen <henrik@brixandersen.dk>

henrikbrixandersen committed:
boards: xtensa: add Olimex ESP32-EVB board definition

Add board definition for the Olimex ESP32-EVB development board.

Signed-off-by: Henrik Brix Andersen <henrik@brixandersen.dk>

henrikbrixandersen committed:
tests: drivers: spi: loopback: add olimex_esp32_evb configuration

Add configuration for testing SPI loopback on the Olimex ESP32-EVB
board.

Signed-off-by: Henrik Brix Andersen <henrik@brixandersen.dk>

martinjaeger commented on boards/xtensa/olimex_esp32_evb/olimex_esp32_evb-pinctrl.dtsi:
```diff
@@ -0,0 +1,56 @@
+		pinmux = <UART1_RX_GPIO36>;
+	};
+
+	i2c0_sda_gpio16: i2c0_sda_gpio16 {
+		pinmux = <I2C0_SDA_GPIO16>;
+		drive-open-drain;
+		output-high;
+	};
+
+	i2c0_scl_gpio13: i2c0_scl_gpio13 {
+		pinmux = <I2C0_SCL_GPIO13>;
+		drive-open-drain;
+		output-high;
+	};
```
SDA and SCL are swapped, if I interpret the schematic correctly.
```suggestion
	i2c0_sda_gpio13: i2c0_sda_gpio13 {
		pinmux = <I2C0_SDA_GPIO13>;
		drive-open-drain;
		output-high;
	};

	i2c0_scl_gpio16: i2c0_scl_gpio16 {
		pinmux = <I2C0_SCL_GPIO16>;
		drive-open-drain;
		output-high;
	};
```

henrikbrixandersen replied:
@martinjaeger Thank you for the thorough review! They were indeed swapped (which is kinda of funny since I had them correct in the first iteration, then - after having probed the signals using a logic analyser - somehow came to the conclusion that they needed to be swapped. Sigh).

martinjaeger commented on boards/xtensa/olimex_esp32_evb/olimex_esp32_evb.dts:
```diff
@@ -0,0 +1,163 @@
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+	sda-pin = <16>;
+	scl-pin = <13>;
+	pinctrl-0 = <&i2c0_sda_gpio16 &i2c0_scl_gpio13>;
```
```suggestion
	pinctrl-0 = <&i2c0_sda_gpio13 &i2c0_scl_gpio16>;
```

henrikbrixandersen commented:
Fixed the `I2C_0` pin assignments as reported by @martinjaeger. Rebased on `main` to pull in the recent ESP32 linker script changes for verification.

@dcpleung Please re-review.

henrikbrixandersen commented:
@martinjaeger Sorry, had to rebase again - I accidentally left some debug in there.
